### ISSUE TITLE
ERROR: Spine 1.2.18 Timed Out While Processing Available Thread Lock

### ISSUE THREAD

**Issue by gj00354347** [Cacti/cacti#4349]
# Describe the bug

Dear Developers ,

since I Have upgraded cacti to 1.2.18 and spine is also upgraded to 1.2.18 , I have been receiving error in the cacti logs which says " ERROR: Spine Timed Out While Processing Available Thread Lock".

one more point I observed is like spine is getting timed out twice at the completion of each poll , so may be spine is triggering 2 processes and bot are getting timed out . Ideally it should spawn single process not 2.

2021-07-19 12:22:06 - SPINE: Poller[1] PID[94822] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:22:06 - SPINE: Poller[1] PID[94850] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:27:06 - SPINE: Poller[1] PID[99494] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:27:06 - SPINE: Poller[1] PID[99470] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:32:04 - SPINE: Poller[1] PID[103658] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:32:04 - SPINE: Poller[1] PID[103675] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:37:06 - SPINE: Poller[1] PID[116028] ERROR: Spine Timed Out While Processing Available Thread Lock
2021-07-19 12:37:06 - SPINE: Poller[1] PID[115966] ERROR: Spine Timed Out While Processing Available Thread Lock

This has caused breaks in many graphs although many are populated with data as well . The strange thing to notice here is that poller is not running at its capacity , like there is a 5 Min poll cycle and its getting completed in 80 seconds or so .

2021-07-19 12:18:31 - SYSTEM STATS: Time:87.2378 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0
2021-07-19 12:23:33 - SYSTEM STATS: Time:89.5563 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0
2021-07-19 12:28:29 - SYSTEM STATS: Time:85.1971 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0
2021-07-19 12:33:29 - SYSTEM STATS: Time:86.1372 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0
2021-07-19 12:38:34 - SYSTEM STATS: Time:90.3089 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0
2021-07-19 12:43:32 - SYSTEM STATS: Time:88.3991 Method:spine Processes:5 Threads:10 Hosts:106 HostsPerProcess:22 DataSources:85363 RRDsProcessed:0

## Screenshots
PFA spine 1.2.18 manual run that too when verbosity=5.
[spine-1.2.18-manual-run-verbosity-5.txt](https://github.com/Cacti/cacti/files/6840221/spine-1.2.18-manual-run-verbosity-5.txt)

## Desktop (please complete the following information)

cacti 1.2.18
spine 1.2.18
Red Hat Enterprise Linux Server release 7.8 (Maipo)
mysql  Ver 15.1 Distrib 5.5.65-MariaDB,
memory 32 G
cores =6
processes = 5
thread = 10
devices = 110
data sources = 85363

PFA spine run manually in verbosity=5 for all devices .

yes i see on forum this question and only one answer which says that 1.2.18 is problematic . PFB url 
[https://forums.cacti.net/viewtopic.php?f=2&t=61762] 

and the strange thing is that when  point spine to old spine 1.2 17 , all the error vanishes and everything is smoooth the the same setting of process and thread .

Best Regards,
Gopal

**Comment by eschoeller** [comment]
Yeah I am seeing issues with 1.2.18 Cacti and Spine. I downgraded spine to 1.2.17. See this issue for spine is probably related:
https://github.com/Cacti/spine/issues/219

**Comment by netniV** [comment]
There are a few issues with the spine changes that seem to only affect the larger systems.  Unfortunately, things have not been working well to grant me time to investigate though.  Downgrading to 1.2.17 is the optimal choice at the moment.

**Comment by TheWitness** [comment]
Moving this to a spine issue.
